extensions/vscode/src/granite/ollama/modelUpdater.ts

```diff
@@ -0,0 +1,187 @@
+      return;
+    }
+
+    const downloadSize = (
+      await Promise.all(
+        modelsToUpdate.map(async (model) => {
+          const modelInfo = await state.ollamaServer.getModelInfo(model);
+          return modelInfo?.size || 0;
+        }),
+      )
+    ).reduce((total, modelSize) => total + modelSize, 0);
+
+    console.log("Models requiring update:", modelsToUpdate);
+    const userChoice = await window.showInformationMessage(
+      `Granite.Code model updates available. Update requires ${formatSize(downloadSize)} download.`,
+      USER_OPTIONS.UPDATE,
+      USER_OPTIONS.CANCEL,
+    );
+
+    if (userChoice === USER_OPTIONS.UPDATE) {
+      await updateModels(state.ollamaServer, modelsToUpdate);
+    }
+  } catch (error) {
+    console.error("Error during model update check:", error);
+  }
+}
+
+async function updateModels(
+  ollamaServer: OllamaServer,
+  models: string[],
+): Promise<void> {
+  return window.withProgress(
+    {
+      location: ProgressLocation.Notification,
+      title: "Updating Granite.Code models",
+      cancellable: true,
+    },
+    async (progress, token) => {
+      const abortController = new AbortController();
+      token.onCancellationRequested(() => {
+        console.log("Model update cancelled by user");
+        abortController.abort();
+      });
+      const progressWrapper = (data: ProgressData): void => {
+        const completed = data.completed ?? 0;
+        const total = data.total ?? 0;
+        let message = data.status;
+
+        if (total > 0) {
+          const progressPercent = Math.round((completed / total) * 100);
+          message = `${message} ${progressPercent}%`;
+        }
+        const increment = data.increment ? (data.increment / total) * 100 : 0;
+        progress.report({ increment, message });
+      };
+
+      try {
+        await ollamaServer.pullModels(
+          models,
+          abortController.signal,
+          progressWrapper,
+        );
+      } catch (error) {
+        if (!abortController.signal.aborted) {
+          console.error("Failed to update models:", error);
+          window.showErrorMessage(
+            "Failed to update models. Please try again later.",
+          );
+        }
+      }
+    },
+  );
+}
+
+async function configureModelUpdateCheck(state: UpdaterState): Promise<void> {
+  const checkUpdates = workspace
+    .getConfiguration(EXTENSION_NAME)
+    .get<boolean>("checkModelUpdates", true);
+
+  if (checkUpdates) {
+    if (!state.updateInterval) {
+      await enableModelUpdateCheck(state);
+    }
+  } else {
+    disableModelUpdateCheck(state);
+  }
+}
+
+export async function registerModelUpdateProvider(
+  context: ExtensionContext,
+): Promise<void> {
+  const state: UpdaterState = {
+    ollamaServer: new OllamaServer(context),
+    checkNow: () => {
+      //Don't check now if onboarding is not completed
+      return !context.globalState.get(SHOW_GRANITE_ONBOARDING_CARD_KEY, true);
```

Comment:
Everywhere else, this is:

   context.globalState.get(SHOW_GRANITE_ONBOARDING_CARD_KEY, true) && !isTestMode

If the intent is "disable it if onboarding is not completed *and* not for test mode", that should be explicit. Should we encapsulate this check? It's already duplicated in 2-3 places.

Reply:
done